### PR TITLE
Feature/migrate to jhipster 6.5.x

### PR DESCRIPTION
**Pull request by kkyrychenko**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -4,52 +4,76 @@ buildscript {
...
+
+apply from: "gradle/docker.gradle"
+apply from: "gradle/sonar.gradle"
 apply plugin: 'war'
-apply plugin: 'propdeps'
-apply plugin: 'io.spring.dependency-management'
-apply plugin: 'idea'
 apply plugin: 'checkstyle'
 apply plugin: 'pmd'
+apply plugin: 'com.github.spotbugs'
```
Comment (v-kyrychenko) [inline build.gradle]: apply plugin it's a old style, better to use id "war" in plugins section

File: build.gradle [hunk 2]
```diff
@@ -60,73 +84,79 @@ pmd {
...
+war {
+
+}
+
+springBoot {
+    mainClassName = "com.icthh.xm.ms.scheduler.SchedulerApp"
+}
+
 test {
-    exclude '**/CucumberTest*'
```
Comment (v-kyrychenko) [inline build.gradle]: this line must be transferred to new version

File: build.gradle [hunk 3]
```diff
@@ -60,73 +84,79 @@ pmd {
...
+    useJUnitPlatform()
+    exclude "**/*IT*", "**/*IntTest*"
 
+    testLogging {
+        events 'FAILED', 'SKIPPED'
+    }
     // uncomment if the tests reports are not generated
     // see https://github.com/jhipster/generator-jhipster/pull/2771 and https://github.com/jhipster/generator-jhipster/pull/4484
     // ignoreFailures true
-    reports.html.enabled = true
```
Comment (v-kyrychenko) [inline build.gradle]: in old version this parameter is true

File: gradle.properties [hunk 4]
```diff
@@ -2,25 +2,40 @@ rootProject.name=scheduler
...
+apt_plugin_version=0.21
+liquibase_plugin_version=2.0.1
+sonarqube_plugin_version=2.8
+spring_no_http_plugin_version=0.0.3.RELEASE
+checkstyle_version=8.24
+
 lombok_version=1.18.2
-xm_commons_version=2.0.21
+xm_commons_version=2.1.0
+spotbugs_version=3.0.0
```
Comment (v-kyrychenko) [inline gradle.properties]: move this line to the " gradle plugin version" section

File: qa/checkstyle.xml [hunk 5]
```diff
@@ -40,7 +40,7 @@
         </module>
         <module name="NeedBraces"/>
         <module name="LeftCurly">
-            <property name="maxLineLength" value="100"/>
+<!--            <property name="maxLineLength" value="100"/>-->
```
Comment (v-kyrychenko) [inline qa/checkstyle.xml:44]: please leave the comment, about the reason why this line was commented (reason was described in the migration document)

File: src/main/jib/entrypoint.sh [hunk 6]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
```
Comment (v-kyrychenko) [inline src/main/jib/entrypoint.sh]: this file and folder must be deleted.

in old version of the project file src/main/docker/entrypoint.sh is used

File: src/main/resources/config/application-prod.yml [hunk 7]
```diff
@@ -59,6 +59,9 @@ spring:
             hibernate.cache.use_query_cache: false
             hibernate.generate_statistics: false
             hibernate.cache.use_minimal_puts: true
+            hibernate.dialect: io.github.jhipster.domain.util.FixedPostgreSQL82Dialect
```
Comment (v-kyrychenko) [inline src/main/resources/config/application-prod.yml]: this is unnecessary properties, this information already defined in the section

```
jpa:
        database-platform: 
```

File: src/main/resources/config/application-tls.yml [hunk 8]
```diff
@@ -0,0 +1,19 @@
+# ===================================================================
```
Comment (v-kyrychenko) [inline src/main/resources/config/application-tls.yml]: delete this new file

File: src/test/resources/config/application.yml [hunk 9]
```diff
@@ -52,6 +52,9 @@ spring:
             hibernate.cache.use_query_cache: false
             hibernate.generate_statistics: true
             hibernate.hbm2ddl.auto: validate
+            hibernate.model.generator_name_as_sequence_name: false # see org.hibernate.cfg.AvailableSettings.PREFER_GENERATOR_NAME_AS_DEFAULT_SEQUENCE_NAME
```
Comment (v-kyrychenko) [inline src/test/resources/config/application.yml:61]: I think this properties also must be added to 
src/resources/config/application.yml

need to check issue with sequence in dev and prod profile

File: src/test/resources/i18n/messages_en.properties [hunk 10]
```diff
@@ -0,0 +1,4 @@
+email.test.title=test title
```
Comment (v-kyrychenko) [inline src/test/resources/i18n/messages_en.properties]: this new file must be deleted

**Review by v-kyrychenko — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 3 above)
Comment (kkyrychenko) [reply build.gradle]: done

File: gradle.properties (hunk 4 above)
Comment (kkyrychenko) [reply gradle.properties]: done

File: qa/checkstyle.xml (hunk 5 above)
Comment (kkyrychenko) [reply qa/checkstyle.xml:44]: done

File: src/main/jib/entrypoint.sh (hunk 6 above)
Comment (kkyrychenko) [reply src/main/jib/entrypoint.sh]: done

File: src/main/resources/config/application-tls.yml (hunk 8 above)
Comment (kkyrychenko) [reply src/main/resources/config/application-tls.yml]: done

File: src/test/resources/i18n/messages_en.properties (hunk 10 above)
Comment (kkyrychenko) [reply src/test/resources/i18n/messages_en.properties]: done

File: src/test/resources/config/application.yml (hunk 9 above)
Comment (kkyrychenko) [reply src/test/resources/config/application.yml:61]: already added to dev, prod

File: src/main/resources/config/application-prod.yml (hunk 7 above)
Comment (kkyrychenko) [reply src/main/resources/config/application-prod.yml]: done

File: build.gradle (hunk 1 above)
Comment (kkyrychenko) [reply build.gradle]: done

File: build.gradle (hunk 2 above)
Comment (kkyrychenko) [reply build.gradle]: done

**Review by v-kyrychenko — DISMISSED** [review]

File: .gitignore [hunk 11]
```diff
@@ -50,6 +50,8 @@ local.properties
 # PDT-specific
 .buildpath
 
+# STS-specific
```
Comment (v-kyrychenko) [inline .gitignore:53]: can't see information about real test execution. Please check this. Expected something like this TestUnitTest > testname PASSED

Comment (kkyrychenko) [reply .gitignore:53]: fixed

**Review by v-kyrychenko — APPROVED** [review]

**Review by amedvedchuk — APPROVED** [review]

File: src/main/java/com/icthh/xm/ms/scheduler/service/TaskQueryService.java [hunk 12]
```diff
@@ -47,6 +48,7 @@
      */
     @FindWithPermission("TASK.GET_LIST")
     @Transactional(readOnly = true)
+    @PrivilegeDescription("Privilege to get all the tasks which matches the criteria from the database")
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/ms/scheduler/service/TaskQueryService.java]: scheduled tasks

File: src/main/java/com/icthh/xm/ms/scheduler/service/TaskQueryService.java [hunk 13]
```diff
@@ -63,6 +65,7 @@
...
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/ms/scheduler/service/TaskQueryService.java]: scheduled tasks

File: src/main/java/com/icthh/xm/ms/scheduler/web/rest/TaskResource.java [hunk 14]
```diff
@@ -67,6 +68,7 @@ public TaskResource(TaskService taskService, TaskQueryService taskQueryService)
     @PreAuthorize("hasPermission({'task': #taskDTO}, 'TASK.CREATE')")
     @PostMapping("/tasks")
     @Timed
+    @PrivilegeDescription("Privilege to create a new task")
     public ResponseEntity<TaskDTO> createTask(@Valid @RequestBody TaskDTO taskDTO) throws URISyntaxException {
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/ms/scheduler/web/rest/TaskResource.java:72]: Privilege to schedule a new task

**Pull request merged**